DMTN-148.tex

```diff
@@ -50,9 +50,276 @@
+  \verb|{OBSTYPE}_SCHEMA| & Creation & Storage schema for calibration.  \\
+  \verb|{OBSTYPE}_VERSION| & Creation & Storage schema version. \\
+  \verb|INSTRUME| & Creation & Instrument/camera this calibration belongs to. \\
+  \verb|DETECTOR| & Creation & Integer identifier for the ccd/detector. \\
+  \verb|FILTER| & Creation & Name of the physical filter or ``NONE''. \\
+  \verb|CALIB_ID| & Creation & DataId that can be used to request this calibration after ingest. \\
+  \verb|CALIB_CREATION_DATE| & Creation & Creation date of the calibration. \\
+  \verb|CALIB_CREATION_TIME| & Creation & Creation time of the calibration. \\
+  \verb|CALIBDATE| & Creation & Concatenation of \verb|CALIB_CREATION_DATE| and \verb|CALIB_CREATION_TIME|. \\
+  \verb|VALIDSTART| & Certification & Start date the calibration is valid from.  \\
+  \verb|VALIDEND| & Certification & End date the calibration is valid from.  \\
+\end{tabular}
+
+
+\subsection{Calibration Construction in Gen3}
+
+Calibration products should be constructed using well defined
+PipelineTask code.  This removes the need to delineate between
+``calibrations'' and ``user-curated calibrations''.  The only
+remaining difference between these two groups is what the initial
+persisted data type is: ``calibrations'' are stored in FITS images, as
+they represent corrections that are only meaningful as full images;
+``user-curated calibrations'' can be visualized in a small number of
+parameters, and so persisting these in a text format to allow user
+inspection is reasonable.  ECSV and YAML are good formats for these
+calibrations, and a calibration's readText/writeText methods should
+implement one of these.
+
+In addition to the calibration product, the PipelineTask code must
+persist a record of the input data used for construction.  This, along
+with the already persisted pipeline definition and task configuration,
+allows the calibration to be regenerated by any user, making the
+construction process much more transparent than it currently is.  A
+YAML file containing the dataId parameters for all inputs would
+satisfy this.  Appending additional construction statistics (for
+example, weight factors applied to each input during the combination
+stage) would be helpful in the validation process.
+
+Newly generated calibration products will belong to a Gen3 collection,
+and that collection will be used to manage the calibration through the
+validation and certification process.  This collection may only
+contain one entry for a given OIDF set.
+
+\subsection{Validation}
+
+Calibration products are not currently validated, largely due to the
+opaque construction process.  Some set of proper validation tasks must
+be made, applying the newly constructed product to a set of test
+exposures prior to that calibration product's certification.  Applying
+these validation tasks continuously as new data is taken will inform
+the date ranges for which the product is certified.
```

Comment:
Is validation best/only done by applying to test exposures, or are there metrics that can be derived solely from the calibration product?  Are the test exposures special in some way?  What "new data" is meant here?  Is that new raw calibration data or new science data?

Reply:
We expect to be able to validate using the new data, and by comparison to the old calibration data.  I don't think we'll need test exposures (indeed, if we did we'd include them in the original calibration sequence;  for example, we may hold back some e.g. bias exposures for validation)

Reply:
I had anticipated validating based on the inputs (ensure the residuals on all inputs are consistent), based on a held back validation sample (ensure that the inputs are representative of all possible inputs), and finally on new calibration data (ensure that the calibration is stable over time).